src/res_core.ml

```diff
@@ -6250,9 +6263,29 @@ and parseAttribute p =
           ] )
   | _ -> None
 
+and parseAttributeForFields p =
+  let isAttributeForField p =
+    Parser.lookahead p (function state ->
```

Comment:
I guess the grammar forces us into using a lookahead?

Reply:
It's super overloaded over there. Apparently `{JSX stufff}` only works because the `JSX` attribute drives the parser into a specific path (which does not consider records). Which itself is arbitrary.

I'm thinking of considering proper surface syntax early on, so one does not need to distinguish based on the name of attributes, which is really not robust.

Reply:
Before this PR, it already distinguished based on the *presence* of attributes, which is also unclear whether intended.